src/topology/vector_bundle.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Nicolò Cavalleri.
+-/
+
+import topology.topological_fiber_bundle
+import analysis.calculus.deriv
+import linear_algebra.dual
+
+/-!
+# Topological vector bundles
+
+In this file we define topological vector bundles.
+
+I believe it is an enough important concept that it should deserve to be a class, unlike fiber
+bundles. The most important idea here is that vector bundles are named through their fibers.
+Let `B` be the base space. The collection of the fibers is a function `E : B → Type*` for which there
+is an appropriate instance on each `E x` and an instance of topological space over `Σ x, E x`.
+Naming conventions are essential to work with vector bundles this way.
+-/
+
+notation V `ᵛ` R := module.dual R V
+
+variables {B : Type*}
+
+def total_space (E : B → Type*) := Σ x, E x
+
+def vector_bundle.proj (E : B → Type*) : total_space E → B := λ y : (total_space E), y.1
+
+variables (R : Type*) (E : B → Type*) (F : Type*)
+  [comm_semiring R] [topological_space B] [∀ x, add_comm_monoid (E x)] [∀ x, topological_space (E x)]
+  [∀ x, semimodule R (E x)]
+  [topological_space F] [add_comm_monoid F] [semimodule R F]
+  [topological_space (total_space E)]
+
+open vector_bundle
+
+/-@[reducible] def dual := (Σ x, (E x)ᵛR)-/
+
+instance is_this_a_good_idea {x : B} : has_coe (E x) (total_space E) := ⟨λ y, (⟨x, y⟩ : total_space E)⟩
```

Comment:
handy but not sure it is a good idea

Reply:
If this is desirable, this should probably be a `has_coe_t` instance, but I'm not sure if even that is a good idea. [Note [use has_coe_t]](https://leanprover-community.github.io/mathlib_docs/notes.html#use%20has_coe_t) doesn't exactly apply, but I think it will lead to the same problems: I'm worried that Lean will try to coerce `E` into `total_space (λ _, E)`.

Reply:
Ok thanks for the comment! My plan is to try to leave it until it creates problem and remove it when problems will arise if they will ever arise, I will also try to change it to `has_coe_t`